### ISSUE TITLE
After some days running well my subscription goes down and works again only after a restart 

### ISSUE THREAD

**Issue by nxtbr** [nats-io/go-nats#350]
Hello guys,

I have 8 worldwide servers and 1 load-balacing api. Our server is a small VPS running Ubuntu. At each server we have an small service written in C# to publish an info about it's network bandwith usage in Mbps each 2.5s. So, our load-balacing, C# too, subcribe NATS and get the servers load and redirect the clients.

After some days running well the subscription is marked as slow and works again only after a restart.

As a noob with NATS probably I am doing something wrong or I am missing something.

What I did?

At each server:

1) when start the service
        _opt = NATS.Client.ConnectionFactory.GetDefaultOptions();

        _opt.Url = config.SyncServerUrl;
        _opt.User = config.SyncServerUser;
        _opt.Password = config.SyncServerPass;
        _opt.AllowReconnect = true;
        _opt.ReconnectWait = config.SyncServerReconnectWait;

        _natsConn = new NATS.Client.ConnectionFactory().CreateConnection(_opt);
2) at each 2s
        if(_natsConn.State == NATS.Client.ConnState.CONNECTED)
        {
                _natsConn.Publish(config.SyncServerKey, data);
        }

At load-balancing

        _opt = NATS.Client.ConnectionFactory.GetDefaultOptions();

        _opt.Url = config.SyncServerUrl;
        _opt.User = config.SyncServerUser;
        _opt.Password = config.SyncServerPass;
        _opt.AllowReconnect = true;
        _opt.ReconnectWait = config.SyncServerReconnectWait;

        _cnn = new NATS.Client.ConnectionFactory().CreateConnection(opt);
        _sub = _cnn.SubscribeAsync(config.SyncServerKey);

        _sub.MessageHandler += S_MessageHandler;
        _sub.Start();
        
        _opt.DisconnectedEventHandler += (sender, ags) => {
                Console.WriteLine("RelaySyncTask >>> Disconnected >>>" + DateTime.UtcNow);

                if (_sub != null) {
                        _sub.MessageHandler -= S_MessageHandler;

                        if (_sub.IsValid) _sub.Unsubscribe();
                }
        };

        _opt.ReconnectedEventHandler += (sender, args) => {
                _sub = _cnn.SubscribeAsync(config.SyncServerKey);

                _sub.MessageHandler += S_MessageHandler;
                _sub.Start();
        };

Other questions:

1) What's the execution flow of theses events (ie.: when a client become slow and the server close its connection is raises Disconnect > Close > AsyncError):

        - Disconnected
        - Closed
        - Reconnected
        - AsyncError

2) When the reconnection happens? And what happen if it fail?

Thanks for the help and keep going the awesome work.

**Comment by ColinSullivan1** [comment]
This is a duplicate of https://github.com/nats-io/csharp-nats/issues/205, so I'm closing this.
